include/motor-control/core/motor_interrupt_handler.hpp

```diff
@@ -92,8 +136,19 @@ class MotorInterruptHandler {
         return bool((old_position ^ position_tracker) & tick_flag);
     }
 
+    [[nodiscard]] auto has_messages() const -> bool {
+        return queue.has_message_isr();
+    }
+    [[nodiscard]] auto can_step() const -> bool {
+        /*
+         * A motor should only try to take a step when the current position
+         * does not equal the target position.
+         */
+        return tick_count < buffered_move.duration;
+    }
+
     void update_move() {
-        has_active_move = queue->try_read_isr(&buffered_move);
+        has_active_move = queue.try_read_isr(&buffered_move);
```

Comment:
Why did we decide to not make the queue a pointer?

Reply:
Now that the queues are provided to the object's constructor, they will always be present and never be null. That makes references (the c++ language feature) the better way to represent them - when you're choosing a reference (the general programming language concept) type in c++ and you're not affected by allocation details, the choice of whether to use a c++ ref or a pointer basically boils down to "do I have to handle this not being present? pointer. will it always be present? reference."